src/front.coffee

```diff
@@ -1,28 +1,72 @@
+moment = require 'moment'
+AbstractAPIAdapter = require 'hubot-abstract-api-adapter'
 try
-	{ Robot, Adapter, TextMessage, User } = require 'hubot'
+	{ TextMessage } = require 'hubot'
 catch
 	prequire = require 'parent-require'
-	{ Robot, Adapter, TextMessage, User } = prequire 'hubot'
-
-class Front extends Adapter
+	{ TextMessage } = prequire 'hubot'
 
+class Front extends AbstractAPIAdapter
 	constructor: ->
-		super
-		@robot.logger.info 'Constructor'
+		try
+			super
+		catch error
+			if error not instanceof TypeError
+				throw error
+
+	extractNext: (obj) -> obj._pagination?.next
+
+	extractResults: (obj) -> obj._results
 
-	send: (envelope, strings...) ->
-		@robot.logger.info 'Send'
+	poll: =>
+		@getUntil(
+			@getOptions process.env.HUBOT_FRONT_API_URL + '/events'
+			@processMessage
+			(obj) => obj.emitted_at > ( @robot.brain.get('AdapterFrontLastKnown') ? moment().subtract(parseInt(process.env.HUBOT_FRONT_SYNC_HISTORY), 'h').unix() )
```

Comment:
It would be nicer to do the `parseInt(process.env.HUBOT_FRONT_SYNC_HISTORY)` just once and store it in a module level constant